Public/Src/Utilities/Native/IO/Windows/FileSystem.Win.cs

```diff
@@ -3004,11 +3004,14 @@ public unsafe uint GetHardLinkCountByHandle(SafeFileHandle handle)
         }
 
         /// <inheritdoc />
-        public FileAttributes GetFileAttributes(string path)
+        public FileAttributes GetFileAttributes(string path, bool throwOnFailure = true)
         {
             if (!TryGetFileAttributes(path, out FileAttributes attributes, out int hr))
             {
-                ThrowForNativeFailure(hr, "FindFirstFileW", nameof(GetFileAttributes));
+                if (throwOnFailure)

```

Comment:
>if (throwOnFailure) [](start = 16, length = 19)

We have TryGetFileAttributes, so we don't need condition whether to throw or not. If you follow my suggestion before, you don't need this change.

Reply:
TryGetFileAttributes is private. With the suggestion above we don't need it, without changing the interface it was necessary.